### PR TITLE
Add Snowflake online store

### PR DESCRIPTION
**Pull request by sfc-gh-madkins**
**What this PR does / why we need it**:

**Which issue(s) this PR fixes**:

Fixes #

**Does this PR introduce a user-facing change?**:

```release-note

```

### CODE REVIEW

**Comment by woop** [comment]
Thanks @sfc-gh-madkins! Please remember to add a description and release note :)

File: sdk/python/tests/integration/feature_repos/repo_configuration.py [hunk 1]
```diff
@@ -46,6 +46,16 @@
...
 REDIS_CONFIG = {"type": "redis", "connection_string": "localhost:6379,db=0"}
+SNOWFLAKE_CONFIG = {
+    "type": "snowflake.online",
+    "account": os.environ["SNOWFLAKE_CI_DEPLOYMENT"],
+    "user": os.environ["SNOWFLAKE_CI_USER"],
+    "password": os.environ["SNOWFLAKE_CI_PASSWORD"],
+    "role": os.environ["SNOWFLAKE_CI_ROLE"],
+    "warehouse": os.environ["SNOWFLAKE_CI_WAREHOUSE"],
+    "database": "FEAST",
+    "schema": "ONLINE",
```
Comment (sfc-gh-madkins) [inline sdk/python/tests/integration/feature_repos/repo_configuration.py:57]: @adchia for visibility

File: sdk/python/feast/infra/utils/snowflake_utils.py [hunk 2]
```diff
@@ -273,6 +275,175 @@ def create_temporary_sfc_stage(cursor: SnowflakeCursor) -> str:
     return stage_name
 
 
+def write_pandas_binary(
```
Comment (adchia) [inline sdk/python/feast/infra/utils/snowflake_utils.py:278]: this looks like it's also similarly copied heavily from the python connector? What do you need that's custom?

Comment (sfc-gh-madkins) [reply sdk/python/feast/infra/utils/snowflake_utils.py:278]: it converts our entity key and value columns to a binary format

File: sdk/python/feast/type_map.py [hunk 3]
```diff
@@ -497,17 +497,17 @@ def snowflake_python_type_to_feast_value_type(
...
         "float64": ValueType.DOUBLE,
         "int64": ValueType.INT64,
         "uint64": ValueType.INT64,
-        "int32": ValueType.INT32,
-        "uint32": ValueType.INT32,
-        "int16": ValueType.INT32,
-        "uint16": ValueType.INT32,
-        "uint8": ValueType.INT32,
-        "int8": ValueType.INT32,
+        "int32": ValueType.INT64,
```
Comment (woop) [inline sdk/python/feast/type_map.py:504]: What's happening here?

**Pull request closed without merging**